apps/jubmoji-quest/src/hooks/useFetchCards.ts

```diff
@@ -29,6 +30,34 @@ export const useFetchCards = () => {
   );
 };
 
+/**
+ * Get list of the collected cards with all the information
+ * @returns JubmojiCardProps
+ */
+export const useFetchCollectedCards = () => {
+  const { isLoading, data: jubmojiCollectionCards = [] } = useFetchCards();
+  const { isLoading: isLoadingJubmojis, data: jubmojis = [] } = useJubmojis();
+
+  return useQuery(
+    ["collectedCards", jubmojis?.length],
+    async () => {
+      // get all jubmojis collected infos
+      const collectedPubKeys = Object.entries(jubmojis).map(
+        ([_index, { pubKeyIndex }]) => pubKeyIndex
+      );
+
+      const collectedJubmojis =
+        collectedPubKeys.map((pubKeyIndex) => {
+          return getJubmojiCardByPubIndex(jubmojiCollectionCards, pubKeyIndex);
+        }) ?? [];
+      return collectedJubmojis;
+    },
+    {
+      enabled: !isLoading && !isLoadingJubmojis,
+    }
+  );
+};
+
```

Comment:
@vb7401 @AndrewCLu we can use this `useFetchCollectedCards ` whenever we need the list of the collected cards